### PR TITLE
Bearer only client

### LINKED ISSUE #26: Bearer only clients
Now that this plugin supports token introspection, it would be useful to allow setting up bearer token only clients. This would mimic Keycloaks native bearer_only adapter setting. 

Instead of redirecting in case of bad or unexisting token, a 401 HTTP error would be thrown with a WWW-Authenticate header also set.

I am preparing a PR for this issue. Opinions?

### PR DESCRIPTION
**Pull request by lodrantl**
Resolves #26.

### CODE REVIEW

**Comment by Trojan295** [comment]
Could you rebase the PR from the current master? There was a problem with CI, which is fixed now.

**Comment by lodrantl** [comment]
Finnally took the time to rebase and add the tests. Also deployed it to staging env and we'll see how it goes.

**Comment by tsyrjanen** [comment]
Looks good. @phirvone could you also look at this.

File: kong/plugins/oidc/handler.lua [hunk 1]
```diff
@@ -59,9 +59,13 @@ function make_oidc(oidcConfig)
 end
 
 function introspect(oidcConfig)
-  if utils.has_bearer_access_token() then
+  if utils.has_bearer_access_token() or oidcConfig.bearer_only == "yes" then
     local res, err = require("resty.openidc").introspect(oidcConfig)
     if err then
+      if oidcConfig.bearer_only == "yes" then
+        ngx.header["WWW-Authenticate"] = 'Bearer realm="kong",error="' .. err .. '"'
```
[the remark this answers is not in the document]
Comment (lodrantl) [reply kong/plugins/oidc/handler.lua]: Also thought about that, but didn't want to overcomplicate the configuration. But now that I think again, I am all for a configurable realm with a sane default.

**Pull request merged**